bind.go

```diff
@@ -306,6 +306,11 @@ func format(tz *time.Location, scale TimeUnit, v any) (string, error) {
 		}
 		return fmt.Sprintf("[%s]", val), nil
 	case fmt.Stringer:
+		if v := reflect.ValueOf(v); v.Kind() == reflect.Pointer &&
+			v.IsNil() &&
+			v.Type().Elem().Implements(reflect.TypeOf((*fmt.Stringer)(nil)).Elem()) {
+			return "NULL", nil
```

Comment:
I made a similar check using `database/sql` as a reference.
This implementation is also copied `go-sql-driver/mysql`.
Should I write a source code comment mentioning the reference source?

* https://cs.opensource.google/go/go/+/6ba6e72e39ac5be92f604832640c3131ad3d0ade:src/database/sql/driver/types.go;l=217-235
* https://github.com/go-sql-driver/mysql/blob/097fe6e3ad83bbd7c84debe810aec4c4a533bcaa/statement.go#L199-L217